lib/board.rb

```diff
@@ -0,0 +1,63 @@
+class Board
+
+  attr_accessor :cells
+
+  @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+
+  def initialize
+    reset
+  end
+
+  def reset
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+      puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+      puts "-----------"
+      puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+      puts "-----------"
+      puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(pos)
+    @cells[pos.to_i - 1]
+  end
+
+  def full?
+    @cells.all? do |token|
+      token == "X" || token == "O"
+    end
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each do |cell|
+      if cell == "X" || cell == "O"
+        counter += 1
+      end
+    end
+    counter
+  end
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "O" || cell == "X"}
end
```